superset/db_engine_specs/bigquery.py

```diff
@@ -20,16 +20,23 @@
 from typing import Any, Dict, List, Optional, Tuple, TYPE_CHECKING
 
 import pandas as pd
+from flask_babel import gettext as __
 from sqlalchemy import literal_column
 from sqlalchemy.sql.expression import ColumnClause
 
 from superset.db_engine_specs.base import BaseEngineSpec
+from superset.errors import SupersetErrorType
 from superset.utils import core as utils
 
 if TYPE_CHECKING:
     from superset.models.core import Database  # pragma: no cover
 
 
+CONNECTION_DATABASE_PERMISSIONS_REGEX = re.compile(
+    "(?:^|(?<= ))(403|POST|permission)(?:(?= )|$)"
```

Comment:
I think this is too broad, it might catch other errors that are unrelated.

It would be better to catch something like:

```python
"Access Denied: Project User does not have bigquery.jobs.create permission in project (?P<project>.+?)"
```